ChromeExtension/js/background.js

```diff
@@ -92,110 +70,108 @@ function createGeniusCurrentTab() {
+                                if (!domain.startsWith("http://") && !domain.startsWith("https://")){
+                                    domain = "http://" + domain;
+                                }
+                            }
+
+                            newLink = domain + "/" + data.ShortUrl.Code;
+                            chrome.storage.local.set({"lastCreatedLink": newLink});
+                            chrome.storage.local.get(["createdLinks"]).then((r6) => {
+                                var createdLinks = r6.createdLinks;
+                                chrome.storage.local.set({"createdLinks": parseInt(createdLinks) + 1});
+                            });
+
+                            tryHandleSuccess(newLink);
+                        },
+                        function (error) {
+                            var parseError = JSV.parse(error);
+                            var error401 = parseError.ResponseStatus.ErrorCode;
+                            if (error401 == 'AuthenticationException') {
+                                alert('Oops! Those keys don\'t appear to be right. Please double check your API Key and Secret.');
+                            } else {
+                                alert('Hmm.. looks like we\'re having trouble connecting. Try again, or email help@geni.us to let us know.');
+                            }
+                        });
+                    });
                 });
-            }
-
-            if (window.location.href === "chrome-extension://" + chrome.runtime.id + "/alertLoadingInside.html") {
-
-                window.location.href = "alertDoneInside.html";
-
-            }
-        },
-        function (error) {
-            var parseError = JSV.parse(error);
-            var error401 = parseError.ResponseStatus.ErrorCode;
-            if (error401 == 'AuthenticationException') {
-                alert('Oops! Those keys don\'t appear to be right. Please double check your API Key and Secret.');
-            } else {
-                alert('Hmm.. looks like we\'re having trouble connecting. Try again, or email help@geni.us to let us know.');
-            }
-
+            });
         });
+    });
 }
 
-function CreateContentMenus() {
-
-    if (localStorage['defaultGroup'] != null && localStorage['defaultGroup'] != '') {
-        chrome.contextMenus.removeAll()
-        chrome.contextMenus.create({
-            title: 'Create geni.us link from current tab',
-            contexts: ['page'],
-            id: 'child1',
-            onclick: createGeniusCurrentTab
-        });
-
-        chrome.contextMenus.create({
-            title: 'Create geni.us link from selected URL',
-            contexts: ['link'],
-            id: 'child2',
-            onclick: createGeniusCurrentLink
-        });
+chrome.runtime.onMessage.addListener(function (msg, sender, sendResponse) {
+    if (msg.action === "createGeniusLink" && msg.url != undefined && msg.url !== "") {

```

Comment:
like I mentioned above, this "createGeniusLink" function needs access to the DOM and "window" object, so it needs to be kept here, but the actual function that is called when the user presses the context menu is under the serviceworker, which will send a message to this background script.

Reply:
changed